core/viewmodel/src/main/kotlin/com/captures2024/soongan/core/viewmodel/profile/ProfileViewModel.kt

```diff
@@ -107,36 +125,80 @@ constructor(
         )
     }
 
-    override suspend fun handleIntent(intent: Intent) {
+    override fun handleIntent(intent: Intent) {
         when (intent) {
-            Intent.Init -> initSyncData()
+            is Intent.Init -> handleInit()
+            is Intent.RefreshMyGallery -> launch { handleRefreshMyGallery() }
+            is Intent.LoadNextPage -> launch { handleLoadNextPage() }
+            is Intent.OnClickMenu -> handleOnClickMenu()
+            is Intent.OnClickNotification -> handleOnClickNotification()
+            is Intent.OnClickPhoto -> handleOnClickPhoto(intent)
+            is Intent.OnClickRegistrationText -> handleOnClickRegistrationText()
+            is Intent.OnCloseBottomSheet -> handleOnCloseBottomSheet(intent)
+        }
+    }
 
-            Intent.RefreshMyGallery -> fetchProfileGallery(page = 0, isRefreshing = true)
+    private fun handleInit() {
+        launch { collectUserProfile() }
+        launch { fetchProfileGallery(page = 0) }
+    }
 
-            Intent.LoadNextPage -> fetchProfileGallery(page = currentState.nextPage)
+    private suspend fun handleRefreshMyGallery() {
+        fetchProfileGallery(
+            page = 0,
+            isRefreshing = true,
+        )
+    }
 
-            Intent.OnClickMenu -> reduce { copy(isOpenBottomSheet = true) }
+    private suspend fun handleLoadNextPage() {
+        fetchProfileGallery(page = currentState.nextPage)
+    }
 
-            Intent.OnClickNotification -> onClickNotification()
+    private fun handleOnClickMenu() {
+        reduce {
+            copy(
+                isOpenBottomSheet = true,
+            )
+        }
+    }
 
-            is Intent.OnClickPhoto -> postSideEffect(Effect.NavigateToHomePost(intent.postId))
+    private fun handleOnClickNotification() {
+        reduce {
+            copy(hasNotification = false)
+        }
 
-            Intent.OnClickRegistrationText -> postSideEffect(Effect.NavigateToRegistrationPost)
+        postSideEffect(Effect.NavigateToNotification)
+    }
 
-            is Intent.OnCloseBottomSheet -> onCloseBottomSheet(intent)
-        }
+    private fun handleOnClickRegistrationText() {
+        postSideEffect(Effect.NavigateToRegistrationPost)
+    }
+
+    private fun handleOnClickPhoto(intent: Intent.OnClickPhoto) {
+        postSideEffect(Effect.NavigateToHomePost(intent.postId))
     }
 
-    private fun initSyncData() {
-        launch {
-            fetchUserProfile()
+    private fun handleOnCloseBottomSheet(intent: Intent.OnCloseBottomSheet) {
+        reduce {
+            copy(
+                isOpenBottomSheet = false
+            )
         }
-        launch {
-            fetchProfileGallery(page = 0)
+
+        when (intent.outType) {
+            ProfileBtmShtOutType.EDIT -> postSideEffect(Effect.NavigateToEditProfile)
```

Comment:
```suggestion
            ProfileBtmShtOutType.EDIT -> blockGuestModeLogic { postSideEffect(Effect.NavigateToEditProfile) }
```
기획의 게스트 모드 블락 시점을 제가 확인은 못했는데 게스트 프로필은 null인 경우 보여지는 값이라 edit 화면까지 넘어갈 필요가 없지 않을지 생각이 들었습니다.